oak_runtime/src/node/crypto/mod.rs

```diff
@@ -0,0 +1,199 @@
+            "/oak.crypto.OakCrypto/Bind" => sinvoke(|req| self.tink.bind(req), req_data),
+            "/oak.crypto.OakCrypto/Unbind" => sinvoke(|req| self.tink.unbind(req), req_data),
+            "/oak.crypto.OakCrypto/KMSProxy" => sinvoke(|req| self.tink.kms_proxy(req), req_data),
+            "/oak.crypto.OakCrypto/Encrypt" => sinvoke(|req| self.tink.encrypt(req), req_data),
+            "/oak.crypto.OakCrypto/Decrypt" => sinvoke(|req| self.tink.decrypt(req), req_data),
+            "/oak.crypto.OakCrypto/EncryptDeterministically" => {
+                sinvoke(|req| self.tink.encrypt_deterministically(req), req_data)
+            }
+            "/oak.crypto.OakCrypto/DecryptDeterministically" => {
+                sinvoke(|req| self.tink.decrypt_deterministically(req), req_data)
+            }
+            "/oak.crypto.OakCrypto/ComputeMac" => {
+                sinvoke(|req| self.tink.compute_mac(req), req_data)
+            }
+            "/oak.crypto.OakCrypto/VerifyMac" => sinvoke(|req| self.tink.verify_mac(req), req_data),
+            "/oak.crypto.OakCrypto/ComputePrf" => {
+                sinvoke(|req| self.tink.compute_prf(req), req_data)
+            }
+            "/oak.crypto.OakCrypto/Sign" => sinvoke(|req| self.tink.sign(req), req_data),
+            "/oak.crypto.OakCrypto/Verify" => sinvoke(|req| self.tink.verify(req), req_data),
+            _ => Err(rpc_status(
+                Code::NotFound,
+                format!("Unknown method_name: {}", method_name),
+            )),
+        }
+    }
+
+    /// Process a gRPC invocation.
+    fn process_invocation(&mut self, runtime: &RuntimeProxy, invocation: &Invocation) {
+        let grpc_response = match invocation.receive_request(runtime) {
+            Ok(request) => {
+                debug!("{}: invoke method {}", self.node_name, request.method_name);
+                let result = self.invoke_method(&request.method_name, &request.req_msg);
+                match result {
+                    Ok(data) => GrpcResponse {
+                        rsp_msg: data,
+                        status: None,
+                        last: true,
+                    },
+                    Err(grpc_err) => GrpcResponse {
+                        rsp_msg: vec![],
+                        status: Some(grpc_err),
+                        last: true,
+                    },
+                }
+            }
+            Err(error) => GrpcResponse {
+                rsp_msg: vec![],
+                status: Some(rpc_status(
+                    Code::InvalidArgument,
+                    format!("Error reading request: {:?}", error),
+                )),
+                last: true,
+            },
+        };
+        let _ = invocation
+            .send_response(grpc_response, runtime)
+            .map_err(|error| {
+                error!("Couldn't send the gRPC response: {:?}", error);
+            });
+    }
```

Comment:
It would be preferable to print the error via an `if let` instead of using `map_err` only for its side effects.

Reply:
Done.